### PR TITLE
Closes #289: login via OAuth2 fix

### LINKED ISSUE #289: Login via google doesn't work

### PR DESCRIPTION
**Pull request by dbojanek**
The actual fix is adding @Named("OAuth2Page") on OAuth2LoginBackingBean class

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java [hunk 1]
```diff
@@ -33,10 +34,11 @@
  * @author michael
  */
 @ViewScoped
+@Named("OAuth2Page")
 public class OAuth2LoginBackingBean implements Serializable {
 
     private static final Logger logger = Logger.getLogger(OAuth2LoginBackingBean.class.getName());
-    private static final long STATE_TIMEOUT = 1000 * 60 * 15; // 15 minutes in msec
+    private static final long STATE_TIMEOUT = 15 * 60 * 1000; // 15 minutes in msec
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java]: Maybe ` TimeUnit.MINUTES.toMillis(15);`?

File: dataverse-persistence/src/main/resources/Bundle_en.properties [hunk 2]
```diff
@@ -1921,20 +1921,6 @@ citationFrame.banner.closeIcon=Close this message, go to dataset
 citationFrame.banner.countdownMessage= This message will close in 
 citationFrame.banner.countdownMessage.seconds=seconds
 
-# Friendly AuthenticationProvider names
-authenticationProvider.name.builtin=Dataverse
```
Comment (MrK191) [inline dataverse-persistence/src/main/resources/Bundle_en.properties:1925]: Are all those values not used?

File: dataverse-webapp/src/main/webapp/loginpage.xhtml [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
       xmlns:ui="http://java.sun.com/jsf/facelets"
       xmlns:o="http://omnifaces.org/ui"
       xmlns:p="http://primefaces.org/ui"
       xmlns:fn="http://java.sun.com/jsp/jstl/functions"
-      xmlns:jsf="http://xmlns.jcp.org/jsf">
+      xmlns:jsf="http://xmlns.jcp.org/jsf"
+      xmlns:f="http://xmlns.jcp.org/jsf/core"
+>
+
+<!--@elvariable id="OAuth2Page" type="edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2LoginBackingBean"-->
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/loginpage.xhtml]: I don't think you need elvariable, the IDE already has direct reference to the backing bean page.

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java]: Just changed it for aesthetics but you're right.

File: dataverse-persistence/src/main/resources/Bundle_en.properties (hunk 2 above)
Comment (dbojanek) [reply dataverse-persistence/src/main/resources/Bundle_en.properties:1925]: They are, but they were duplicated.

File: dataverse-webapp/src/main/webapp/loginpage.xhtml (hunk 3 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/loginpage.xhtml]: Yeah, you're right. It was needed before I've added @Named. Should  work fine without it now.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java [hunk 4]
```diff
@@ -33,10 +35,11 @@
  * @author michael
  */
 @ViewScoped
+@Named("OAuth2Page")
 public class OAuth2LoginBackingBean implements Serializable {
 
     private static final Logger logger = Logger.getLogger(OAuth2LoginBackingBean.class.getName());
-    private static final long STATE_TIMEOUT = 1000 * 60 * 15; // 15 minutes in msec
+    private static final long STATE_TIMEOUT = TimeUnit.MINUTES.toMillis(15); // 15 minutes in msec
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OAuth2LoginBackingBean.java]: I don't think comment is necessary now.

**Review by MrK191 — APPROVED** [review]

**Pull request merged**